internal/services/shared/schema.go

```diff
@@ -321,11 +321,47 @@ func sanityCheckNamespaceChanges(
 	for _, delta := range diff.Deltas() {
 		switch delta.Type {
 		case nsdiff.RemovedRelation:
+			// NOTE: We add the subject filters here to ensure the reverse relationship index is used
```

Comment:
can't we solve this with an index hint?
also removing this in one commit and adding it back in another one obscures the git blame